### PR TITLE
188445579 update dependencies; node compatibility

### PR DESCRIPTION
**Pull request by kswenson**
[[PT-188445579]](https://www.pivotaltracker.com/story/show/188445579)

- node 22 (latest LTS) compatibility seems to have been fixed by uninstalling/reinstalling `jest-environment-jsdom`
- most code changes related to Sass deprecating `@import` in favor of `@use`
- fixes npm-reported vulnerabilities
- fixes lingering lint warnings

### CODE REVIEW

File: v3/src/components/common/formula-insert-values-menu.tsx [hunk 1]
```diff
@@ -25,46 +26,46 @@ export const InsertValuesMenu = ({setShowValuesMenu}: IProps) => {
   )
   const attributeNames = dataSet?.attributes.map(attr => attr.name)
   const globalManager = dataSet && getGlobalValueManager(getSharedModelManager(dataSet))
-  const globals = globalManager
-                    ? Array.from(globalManager.globals.values()).map(global => ({ label: global.name }))
-                    : []
+  const globals = useMemo(() => globalManager
+    ? Array.from(globalManager.globals.values()).map(global => ({ label: global.name }))
```
Comment (scytacki) [inline v3/src/components/common/formula-insert-values-menu.tsx:30]: If a new global is added to the globalManager I think the `globals` memo will not be updated. That seems like a bug?

**Review by scytacki — CHANGES_REQUESTED** [review]
Mostly looks good, but I think the computation of the globals will be stale in the InsertValuesMenu.
Perhaps the menu component is re-created every time it is shown so the globals will always get recomputed?

File: v3/src/components/common/formula-insert-values-menu.tsx (hunk 1 above)
Comment (kswenson) [reply v3/src/components/common/formula-insert-values-menu.tsx:30]: Good question -- I'll look into it.

**Review by scytacki — CHANGES_REQUESTED** [review]
I think this has the same problem as before. The dataSet object probably does not change when a new global is added. So the getListContainerStyle callback will not get re-created and will be computing its position based on the old list of globals.

**Comment by kswenson** [comment]
>I think this has the same problem as before. The dataSet object probably does not change when a new global is added. So the getListContainerStyle callback will not get re-created and will be computing its position based on the old list of globals.

"Once more unto the breach..." -- I eliminated the caching of `listContainerStyle` altogether.

**Review by scytacki — APPROVED** [review]
Looks good. 👍

**Pull request merged**